src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py

```diff
@@ -363,6 +363,13 @@ def load_arguments(self, _):
                    help="Geographic location where Function App will be hosted. Use 'functionapp list-consumption-locations' to view available locations.")
         c.argument('runtime', help='The functions runtime stack.', arg_type=get_enum_type(set(LINUX_RUNTIMES).union(set(WINDOWS_RUNTIMES))))
         c.argument('os_type', arg_type=get_enum_type(OS_TYPES), help="Set the OS type for the app to be created.")
+        c.argument('create_app_insights', arg_type=get_three_state_flag(return_label=True), help="Creates a new App Insights and adds to the new Function app. Needs --app-insights-location.")
```

Comment:
This is a bit too much to expose 5 more arguments for just one scenario support.
I would  much prefer you just author a command of `az appinsights create` (you can put it inside the appservice module for the time being), and let `functionapp create` take an argument to point to an existing appinsights resource.
CLI community are asking for app insights support, why not start to make contributions here to help other users? 
Once the command is in, we can further add a few more to make CRUD complete.  For reference, check out https://github.com/Azure/azure-cli/issues/5543

Reply:
@ahmedelnably, if we have a global `az appinsights create`, would we still need to add support for users to create appinsights using `az functionapp create`, or could we just allow users to provide an exisiting appinsights (which they can create `az appinsights create`, and make the functionapp point to that? Just like we do with storage.

Reply:
@yugangw-msft, on offline discussion with my team as well as the AppInsights team, we decided that I should not make `az appinsights` to avoid ownership and maintenance issues. Appinsights team do have this work in their backlog, so they should be able to take care of that. Just wanted to post an update here.

I still need to update this PR to avoid introducing the new args.